.github/workflows/publish_website.yml

```diff
@@ -11,21 +11,30 @@ on:
     branches:
       - main
     paths:
```

Comment:
Another idea: as this workflow gets more complicated, we could run the _build_ and not the _publish_ parts on pull requests. (I suppose we could also upload to a staging site automatically too... that'd be neat and could help with code reviews :P)